### DISCUSSION TITLE
How to snapshot all entities?

### DISCUSSION THREAD

**Discussion by hasikill** [skypjack/entt#1160] [Q&A] [answered]
How to snapshot all entities? For network rollback.

**Comment by skypjack** [comment]
Do you mean with the snapshot class? Just use the `entt::entity` type or your own entity type if you're using a custom one.

**Reply by hasikill** [reply]
Suppose I add a lot of entities to the entt::registry, and also add a lot of components to the entity. After running for a while, I want to roll back the entt::registry to a certain point in time before this. What should I do? Can the entt::registry be serialized to a file or another client on the network?

**Reply by skypjack** [reply, marked as answer]
Well, I mean, what's stopping you from doing it? There is a small utility to serialize what you are interested in in the snapshot hpp file, otherwise you can write your own serialization routine based on views, storage classes and so on. As for the destination, a file or a chunk of memory is the same from the point of view of a serializer. There is a test in the project that writes a regsitry in a bunch of stacks to re-read it later on. 🤷‍♂️

**Reply by hasikill** [reply]
Sorry, I skipped this example before. I just learned about this library not long ago. After reading this example, I already know how to do it. Thank you
